sycl/test-e2e/Adapters/large_arg.cpp

```diff
@@ -0,0 +1,40 @@
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+
+// REQUIRES: cuda || hip
+// The CUDA and HIP adapters currently have a maximum size for kernel
+// arguments. This test verifies that the adapters properly check that the
+// kernel arguments aren't overflowing this maximum size.
+
+#include <sycl/sycl.hpp>
+#include <array>
+
+int main() {
+  sycl::queue q;
+
+  std::array<float, 1005> data;
+  data.fill(0);
+  data[0] = 5;
+
+  bool caught = false;
```

Comment:
I'd probably do `bool caught{false};` to be consistent with the style of that file. And maybe stick that `1005` in `constexpr`, but that's nothing more than a NIT.